iothub/service/src/RetryPolicies/IotHubServiceRetryPolicyBase.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public abstract class IotHubServiceRetryPolicyBase : IIotHubServiceRetryPolicy
+    {
+        private readonly Random _rng = new();
+        private readonly object _rngLock = new();
+
+        /// <summary>
+        /// Creates an instance of this class.
+        /// </summary>
+        /// <param name="maxRetries">The maximum number of retries; use a negative value for infinite retries.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Throw if the value of <paramref name="maxRetries"/> is negative.</exception>
+        protected IotHubServiceRetryPolicyBase(uint maxRetries)
+        {
+            if (maxRetries < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maxRetries), maxRetries, string.Format(CultureInfo.CurrentCulture, "ArgumentCannotBeNegative", new object[]
```

Comment:
Same, let's just add support in this library for `Argument.AssertNotNegativeValue(...)`.